polling_stations/apps/data_finder/helpers/geocoders.py

```diff
@@ -65,16 +60,11 @@ def geocode_point_only(self):
 
 
 class AddressBaseGeocoderAdapter(BaseGeocoder):
-    def geocode(self):
+    def geocode(self, uprn=None):
         geocoder = AddressBaseGeocoder(self.postcode)
         geocoder.centroid
-
-        try:
-            geocoder.get_code("lad")
-        except MultipleCodesException as e:
-            # re-raise as a more specific MultipleCouncilsException
-            # because that is what the calling code expects to handle
-            raise MultipleCouncilsException(str(e))
+        # We should always have a uprn so only attempt to get council code with this information.
+        geocoder.get_code("lad", uprn=uprn)
```

Comment:
There's a couple of things that don't make sense here:
1. If we should always have a UPRN to call this, why is uprn an optional param and why is `geocode_point_only()` calling it without one?
2. The reason we were calling `geocoder.get_code("lad")` here at all was to see if it was going to throw `MultipleCodesException` so we could re-throw it as `MultipleCouncilsException`.. but if we call `get_code()` with a UPRN, we can't throw that so we're just calling this for no reason and binning the result. We can also bin the call to `geocoder.centroid` (me-from-the-past pointlessly left this here - whoops). I think we just want this to be

```py
def geocode(self):
    return AddressBaseGeocoder(self.postcode)
```

but then when we call `get_code()` on the geocode object, we always pass it a UPRN.

I've not thought this through 100%, but given we're moving to a model where we can only import addresses which have a UPRN attached to them which exists in AddressBase, I wonder if the model of trying `AddressBaseGeocoder` and fall back to `OnspdGeocoder` if we can't find it is even still the right abstraction for `geocode()` (which is used in the finder, but not the importers)? Would it be better to just look at `AddressBaseGeocoder` and only fall back to `OnspdGeocoder` if the `AddressBaseGeocoder` throws `NorthernIrelandException`? https://democracyclub.github.io/uk-geo-utils/geocoders/#exceptions (this relates to one of today's other comments).